### ISSUE TITLE
Gradle: Execution optimizations have been disabled for task xxx .. Reason: Task ':knit'

### ISSUE THREAD

**Issue by nomisRev** [Kotlin/kotlinx-knit#30]
When using Knit with Gradle, it disables optimizations for important Kotlin tasks.

Below is the warning from Gradle.
I'm not so familiar with Gradle but would be to contribute a fix if someone can guide me in the right direction.

This issue is reproducible here: https://github.com/nomisRev/kotlin-kafka

```
> Task :compileKotlin

Execution optimizations have been disabled for task ':compileKotlin' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/Users/simonvergauwen/Projects/KafkaPlayground/build/classes/kotlin/main'. Reason: Task ':knit' uses this output of task ':compileKotlin' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.3/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/simonvergauwen/Projects/KafkaPlayground/build/kotlin/compileKotlin'. Reason: Task ':knit' uses this output of task ':compileKotlin' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.3/userguide/validation_problems.html#implicit_dependency for more details about this problem.

```

Thanks for this amazing tool!

**Comment by thumannw** [comment]
Should be fixed with #39.

**Comment by qwwdfsad** [comment]
Fixed in 0.4.0
